tests/msc2716_test.go

```diff
@@ -734,6 +837,38 @@ func ensureVirtualUserRegistered(t *testing.T, c *client.CSAPI, virtualUserLocal
 	}
 }
 
+func sendMarkerAndEnsureBackfilled(t *testing.T, as *client.CSAPI, c *client.CSAPI, roomID, insertionEventID string) {
+	t.Helper()
+
+	// Send a marker event to let all of the homeservers know about the
+	// insertion point where all of the historical messages are at
+	markerEvent := b.Event{
+		Type: markerEventType,
+		Content: map[string]interface{}{
+			markerInsertionContentField: insertionEventID,
+		},
+	}
+	// We can't use as.SendEventSynced(...) because application services can't use the /sync API
+	markerSendRes := as.MustDoFunc(t, "PUT", []string{"_matrix", "client", "r0", "rooms", roomID, "send", markerEvent.Type, "txn-m123"}, client.WithJSONBody(t, markerEvent.Content))
```

Comment:
Is it a problem that the txn id never changes?

Reply:
Since the homeserver is new every run, I think it's fine. I did notice that `txn-m123` was re-used once, so I updated to something different.

Might be nice to add a helper function to `client` to get a random unique one easily.